src/libraries/Common/src/Interop/Windows/SspiCli/SecuritySafeHandles.cs

```diff
@@ -318,6 +317,59 @@ public new IntPtr DangerousGetHandle()
 
             return errorCode;
         }
+
+        public static unsafe int AcquireCredentialsHandle(
+            string package,
+            Interop.SspiCli.CredentialUse intent,
+            Interop.SspiCli.SCH_CREDENTIALS* authdata,
+            out SafeFreeCredentials outCredential)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Enter(null, package, intent, (IntPtr)authdata);
+
+            int errorCode = -1;
+            long timeStamp;
+
+            // If there is a certificate, wrap it into an array.
+            // Not threadsafe.
+            IntPtr copiedPtr = authdata->paCred;
+            try
+            {
+                IntPtr certArrayPtr = new IntPtr(&copiedPtr);
+                if (copiedPtr != IntPtr.Zero)
+                {
+                    authdata->paCred = certArrayPtr;
+                }
```

Comment:
I don't understand why we're doing this dance with authdata->paCred to replace it with an address that contains the desired value.  Why not just create the authdata correctly in the first place?

Reply:
That part I don't know. I will investigate. Possibly fixing only on new code path to limit risk for now.

Reply:
removed, see the other comment.